### PR TITLE
Built-ins in the meta-theory and lifting hooked sorts and symbols

### PR DESCRIPTION
**Pull request by hreada**
A new subsection (Section 7.9) about dealing with built-ins in the meta-theory is added.
Based on that, lifting rules for hooked sorts and symbols are added, which are essentially the same as the lifting rules for normal sorts and symbols.

###### Reviewer checklist

- [ ] Test coverage: `stack test --coverage`
- [ ] Public API documentation: `stack haddock`
- [ ] Style conformance: `stylish-haskell`

---

### CODE REVIEW

File: docs/semantics-of-k.tex [hunk 1]
```diff
@@ -4644,6 +4651,49 @@ \subsection{More about Symbols}
...
 
+\subsection{Built-ins in the Meta-Theory}
+
+Recall that in Section~\ref{sec:builtins},
+we define matching logic theories \emph{with built-ins} as a tuple
+$(S_\builtin, \Sigma_\builtin, S, \Sigma, A)$,
+where $S_\builtin$ contains built-in sorts
+and $\Sigma_\builtin$ contains built-in symbols.
+Unlike $S$ and $\Sigma$ whose semantics is axiomatized by
+a set $A$ of axioms, the semantics of built-ins are given
```
Comment (bmmoore) [inline docs/semantics-of-k.tex]: It would be clearer to say "the axiomatization of the semantics of built-ins are given by ...", if that's correct.

Comment (hreada) [reply docs/semantics-of-k.tex]: We don't axiomatize the semantics of built-ins; but I will add some more texts discussing what an oracle is to the document.

Comment (hreada) [reply docs/semantics-of-k.tex]: @bmmoore I've addresse all comments. Please review.

**Comment by hreada** [comment]
@bmmoore I've addresse all comments. Please review.

**Comment by bmmoore** [comment]
Changes look good. Have you heard from @grosu ?

**Pull request merged**